extension/src/providers/gotoProvider.ts

```diff
@@ -0,0 +1,131 @@
+import { utils } from 'mocha';
+import * as vscode from 'vscode';
+import { AutoLispExt } from '../extension';
+import { Sexpression } from '../format/sexpression';
+import { ReadonlyDocument } from '../project/readOnlyDocument';
+import { escapeRegExp } from "../utils";
+import { SearchPatterns, SearchHandlers } from './providerShared';
+
+export class AutolispDefinitionProvider implements vscode.DefinitionProvider{
+	async provideDefinition(document: vscode.TextDocument, position: vscode.Position, token: vscode.CancellationToken): Promise<vscode.Location | vscode.Location[]> {
+		const editor: vscode.TextEditor = vscode.window.activeTextEditor;
+		const rDoc = ReadonlyDocument.getMemoryDocument(document);
+		let selected: string = editor.document.getText(editor.selection);
+		if (selected === '') {
+			rDoc.atomsForest.forEach(sexp => {
+				if (sexp instanceof Sexpression && sexp.contains(position)){
+					const atom = sexp.getAtomFromPos(position);
+					if (atom && !atom.isComment()) {
+						selected = atom.symbol.replace(/^[']*/, '');
+					}
+				}
+			});
+		}
+		if (['(', ')', '\'', '.', ';'].includes(selected)){
+			return;
+		}
+		try {
+			// determine scope
+			const {isFunction, parentContainer} = SearchHandlers.getSelectionScopeOfWork(rDoc, position, selected);
+			const locations: vscode.Location[] = [];
+			if (isFunction) {
+				// This has a "preference" for opened and project documents for actual definitions, but will only handle variables on the opened document.
+				let possible = this.findDefunMatches(selected, AutoLispExt.Documents.OpenedDocuments.concat(AutoLispExt.Documents.ProjectDocuments));
```

Comment:
Maybe another refinement is to do searching in the `current ` file firstly, because most of the definition is in the same file; if not found, it can search `opened, workspace, prj` files

Reply:
- Removed all selection checking and leaned on discovery of position atom so that comment checks would happen automatically.
- I had it check the active document first and foremost before moving onto the other tiers of contextual importance. I am not sure if you intended PRJ to be last in your list or not, but I generally kept opened/PRJ's lumped together because we don't auto-reload Projects into the interface, it is a very clear intent of contextual preference for a PRJ to present.